DESIGN.md

```diff
@@ -325,18 +325,99 @@ In the addition of the previous rules, an operator for `Flowable`:
 
 ### Creation
 
-Creation of the various types should be exposed through factory methods that provide safe construction.
+Unlike RxJava 1.x, 2.x base classes are abstract and stateless and generally no longer wrap an `OnSubscribe` callback - this saves allocation in assembly time without limiting the expressiveness. Operator methods and standard factories still live as final on the base classes. 
+
+Since Java still doesn't have extension methods, "adding" more operators can only happen through helper methods such as `lift(C -> C)` and `compose(R -> P)` where `C` is the default consumer type (i.e., `rs.Subscriber`), `R` is the base type (i.e., `Flowable`) and `P` is the base interface (i.e., `rs.Publisher`). As before, the library itself may gain or lose standard operators and/or overloads through the same community process.
```

Comment:
> "adding" more operators can only happen through

This version of `compose` seems to be very similar to the proposed `extend` method on `Consumable` as suggested in my prototype, as merged on the `Observable.extend` and as we discussed in prior pull requests. Is the intent with the `compose` overload to replace the `extend` method?